client/src/unifyfs.c

```diff
@@ -1129,8 +1153,8 @@ int unifyfs_fid_extend(int fid, off_t length)
     /* TODO: move this statement elsewhere */
     /* increase file size up to length */
     if (meta->storage == FILE_STORAGE_FIXED_CHUNK) {
-        if (length > meta->size) {
-            meta->size = length;
+        if (length > meta->local_size) {
+            meta->local_size = length;
         }
     }
```

Comment:
I think we could drop this logic, so long as we're increasing the log_size elsewhere.

Reply:
I agree, but let's git rid of all the FILE_STORAGE_FIXED_CHUNK code in a separate PR.  This patch is getting pretty big..

Reply:
sure, sounds good